lib/AST/ASTPrinter.cpp

```diff
@@ -3179,11 +3196,18 @@ suppressingFeatureCoroutineAccessors(PrintOptions &options,
 static void
 suppressingFeatureABIAttribute(PrintOptions &options,
                                llvm::function_ref<void()> action) {
-  llvm::SaveAndRestore<bool> scope(options.PrintSyntheticSILGenName, true);
-  unsigned originalExcludeAttrCount = options.ExcludeAttrList.size();
-  options.ExcludeAttrList.push_back(DeclAttrKind::ABI);
+  llvm::SaveAndRestore<bool> scope1(options.PrintSyntheticSILGenName, true);
+  ExcludeAttrRAII scope2(options.ExcludeAttrList, DeclAttrKind::ABI);
+  action();
+}
+
+static void
+suppressingFeatureAddressableTypes(PrintOptions &options,
+                                   llvm::function_ref<void()> action) {
+  llvm::SaveAndRestore<bool> scope1(options.PrintSyntheticSILGenName, true);
```

Comment:
Doesn't this need to check its own option instead of `options.PrintSyntheticSILGenName`?

Reply:
I don't need that line at all, really. It was just a copy-paste error. Thanks for catching it!